tests/saw/spec/handshake/s2n_handshake_io.cry

```diff
@@ -81,6 +86,7 @@ normalizeAfterData hsas = map snd (tail (scanl f (False, dataMsg) hsas))
           dataMsg = mkAct TLS_APPLICATION_DATA 0 'B'
 
 // match actions, but drop any with an 'A' writer from the second trace
+// BUG: We don't have an 'A' writer any longer.
```

Comment:
We should just delete this comment.

Reply:
I'm pretty unclear on this entire block. If we remove the 'A' state comments (which we should), we are left with "match actions". What exactly is happening here?

Reply:
Essentially just "match actions" :) In sequential language, the function takes two sequences of handshake_actions (xs and ys). Until it reaches the end of the sequences (i >= `n, j >= `n), it checks that two items at the same index are equal (xs@i == ys@j), then recursively calls itself on the next pair in the next index (go i+1 j+1).